lib/puppet/application/lookup.rb

```diff
@@ -331,6 +331,25 @@ def generate_scope
     end
 
     node = Puppet::Node.indirection.find(node) unless node.is_a?(Puppet::Node) # to allow unit tests to pass a node instance
+
+    if options[:fact_file]
```

Comment:
Nitpick but given that `options[:fact_file}` is accessed up to four times, wouldn't it be better to first extract it to a `fact_file` variable and avoid repeated hash lookups?